src/olympia/files/tests/test_utils_.py

```diff
@@ -588,6 +591,55 @@ def test_get_all_files_prefix_with_strip_prefix():
     ]
 
 
+def test_atomic_lock_with():
+    lock = flufl.lock.Lock('/tmp/test-atomic-lock1.lock')
+
+    assert not lock.is_locked
+
+    lock.lock()
+
+    assert lock.is_locked
+
+    with utils.atomic_lock('/tmp/', 'test-atomic-lock1') as lock_attained:
+        assert not lock_attained
+
+    lock.unlock()
+
+    with utils.atomic_lock('/tmp/', 'test-atomic-lock1') as lock_attained:
+        assert lock_attained
+
+
+def test_atomic_lock_with_lock_attained():
+    with utils.atomic_lock('/tmp/', 'test-atomic-lock2') as lock_attained:
+        assert lock_attained
+
+
+@mock.patch.object(flufl.lock._lockfile, 'CLOCK_SLOP', timedelta(seconds=0))
+def test_atomic_lock_lifetime():
+    def _get_lock():
+        return utils.atomic_lock('/tmp/', 'test-atomic-lock3', lifetime=1)
+
+    with _get_lock() as lock_attained:
+        assert lock_attained
+
+        lock2 = flufl.lock.Lock('/tmp/test-atomic-lock3.lock')
+
+        with pytest.raises(flufl.lock.TimeOutError):
+            # We have to apply `timedelta` to actually raise an exception,
+            # otherwise `.lock()` will wait for 2 seconds and get the lock
+            # for us. We get a `TimeOutError` because we were locking
+            # with a different claim file
+            lock2.lock(timeout=timedelta(seconds=0))
+
+        with _get_lock() as lock_attained2:
+            assert not lock_attained2
+
+        time.sleep(2)
```

Comment:
I tried doing sleeps that are <1s but the tests were pretty unreliable with that unfortunately :(